front/src/applications/operationalStudies/helpers/upsetNewProjectedTrains.ts

```diff
@@ -0,0 +1,39 @@
+import type { TrainSpaceTimeData } from 'applications/operationalStudies/types';
+import { type ProjectPathTrainResult, type TrainScheduleResult } from 'common/api/osrdEditoastApi';
+import { convertIsoUtcToLocalTime } from 'utils/date';
+import type { Dictionary } from 'utils/types';
+
+// TODO: feed the spaceTimeChart with date instead of formatted string
+const formatDatetimeForSpaceTimeChart = (departureTime: string) =>
+  `${convertIsoUtcToLocalTime(departureTime).slice(0, -6)}Z`;
+
+const upsertNewProjectedTrains = (
+  projectedTrains: Dictionary<TrainSpaceTimeData> | undefined,
+  projectedTrainsToUpsert: Dictionary<ProjectPathTrainResult>,
```

Comment:
Maybe `Map<number, ProjectPathTrainResult>` would be more appropriate here?

Reply:
I used a record :)

Reply:
The problem with `Record` is that keys are strings, which we need to convert around to numbers. `Map` doesn't have this issue. (`Map` also fixes a bunch of issues related to JS objects.)

Reply:
Done, I used Map everywhere I can